pinot-plugins/pinot-minion-tasks/pinot-minion-builtin-tasks/src/main/java/org/apache/pinot/plugin/minion/tasks/MinionTaskUtils.java

```diff
@@ -67,7 +72,35 @@ static PinotFS getOutputPinotFS(Map<String, String> taskConfigs, URI fileURI)
     return PinotFSFactory.create(fileURIScheme);
   }
 
-  static PinotFS getLocalPinotFs() {
+  public static Map<String, String> getPushTaskConfig(String tableName, Map<String, String> taskConfigs,
+      ClusterInfoAccessor clusterInfoAccessor) {
+    try {
+      URI outputDirURI = URI.create(clusterInfoAccessor.getDataDir() + "/" + tableName);
```

Comment:
I think we should check push mode first before deciding the outputDir? 

If push mode is tar, we simply need to build segment locally and tar push to Pinot.

Let me know if I misunderstood something

Reply:
It is not really used in the case when pushMode is TAR.

```java
if (BatchConfigProperties.SegmentPushType.valueOf(pushMode.toUpperCase())
    != BatchConfigProperties.SegmentPushType.TAR) {
  outputSegmentTarURI = moveSegmentToOutputPinotFS(configs, convertedTarredSegmentFile);
  LOGGER.info("Moved generated segment from [{}] to location: [{}]", convertedTarredSegmentFile,
      outputSegmentTarURI);
} else {
  outputSegmentTarURI = convertedTarredSegmentFile.toURI();
}

......
case TAR:
  try (PinotFS pinotFS = MinionTaskUtils.getLocalPinotFs()) {
    SegmentPushUtils.pushSegments(
        spec, pinotFS, Arrays.asList(outputSegmentTarURI.toString()), headers, parameters);
  } catch (RetriableOperationException | AttemptsExceededException e) {
    throw new RuntimeException(e);
  }
  break;
 ```

Reply:
I agree that it is not really used in the case when push mode is TAR. However, I feel the current logic is a little bit hard to follow if we don't have the full context. 

If we can check push mode first then decide what to do, it's easier to follow. WDYT?